Helper/PaymentResponseHandler.php

```diff
@@ -279,6 +290,61 @@ public function handlePaymentsDetailsResponse(
                 break;
             case self::REFUSED:
             case self::CANCELLED:
+                $getGiftcardDetails = $this->hasActiveGiftCardPayments(
+                    $paymentsDetailsResponse['merchantReference']
+                );
+                if (!empty($getGiftcardDetails)) {
+                    //Cancel the Authorised Payments
+                    $storeId = $order->getStoreId();
+                    $client = $this->dataHelper->initializeAdyenClient($storeId);
+                    $service = $this->dataHelper->initializeOrdersApi($client);
+                    foreach ($getGiftcardDetails as $giftcardData) {
+                        try {
+                            // Decode JSON response and validate it
+                            $response = json_decode($giftcardData['response'], true);
+                            if (json_last_error() !== JSON_ERROR_NONE || !isset($response['order'])) {
+                                throw new InvalidArgumentException('Invalid giftcard response data');
+                            }
+
+                            // Extract order data and PSPRef
+                            $orderData = $response['order']['orderData'] ?? null;
+                            $pspReference = $response['order']['pspReference'] ?? null;
+
+                            if (!$orderData || !$pspReference) {
+                                throw new InvalidArgumentException('Missing orderData or pspReference in the response');
+                            }
+
+                            // Prepare cancel request
+                            $merchantAccount = $this->configHelper->getAdyenAbstractConfigData("merchant_account", $storeId);
+                            $cancelRequest = [
+                                'order' => [
+                                    'pspReference' => $pspReference,
+                                    'orderData' => $orderData,
+                                ],
+                                'merchantAccount' => $merchantAccount,
+                            ];
+
+                            // Call the cancel service
+                            $cancelResponse = $service->cancelOrder(new CancelOrderRequest($cancelRequest));
```

Comment:
Could you please log the request and response of `/orders/cancel` endpoint ([example](https://github.com/Adyen/adyen-magento2/blob/b5309b0c0ac0aff9755c82ee23a2e0e88dde5ce7/Gateway/Http/Client/TransactionPayment.php#L149))?